product_supplierinfo_for_customer/models/product_supplierinfo.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class ProductSupplierinfo(models.Model):
+    _inherit = 'product.supplierinfo'
+
+    type = fields.Selection(
+        selection=[
+            ('customer', 'Customer'),
+            ('supplier', 'Supplier'),
+        ], string='Type',
+        default='supplier')
+
+    @api.multi
+    @api.onchange('type')
+    def onchange_type(self):
+        if self.type == 'supplier':
```

Comment:
this is broken due to multi, and will fail if you have recordset of >1 values

Also, as you use this onchange only in tests it is better to leave a comment about that in the method as it is confusing )

Reply:
I will just delete `@api.multi`